src/Microsoft.OData.Core/UriParser/Parsers/SelectExpandOptionParser.cs

```diff
@@ -427,7 +408,7 @@ private List<ExpandTermToken> BuildStarExpandTermToken(PathSegmentToken pathToke
             }
 
             // As 2016/1/8, the navigation property is only supported in entity type, and will support in ComplexType in future.
-            var entityType = parentEntityType as IEdmEntityType;
+            var entityType = this.parentStructuredType as IEdmEntityType;

```

Comment:
shouldn't we support ComplexType here, and not just IEdmEntityType?

Reply:
It's a different feature gap.  Let's file an issue to fix it. [https://github.com/OData/odata.net/issues/1493](https://github.com/OData/odata.net/issues/1493)   

---
In reply to: [296959088](https://github.com/OData/odata.net/pull/1488#discussion_r296959088) [](ancestors = 296959088)